test/e2e/fpga/fpga.go

```diff
@@ -0,0 +1,143 @@
+		framework.RunKubectlOrDie("delete", "daemonset", "intel-fpga-plugin", "--namespace", "kube-system")
+	})
+
+	serviceAccountYamlPath, err := utils.LocateRepoFile(serviceAccountYaml)
+	if err != nil {
+		framework.Failf("unable to locate %q: %v", serviceAccountYaml, err)
+	}
+
+	pluginYamlPath, err := utils.LocateRepoFile(pluginYaml)
+	if err != nil {
+		framework.Failf("unable to locate %q: %v", pluginYaml, err)
+	}
+
+	webhookDeployScriptPath, err := utils.LocateRepoFile(webhookDeployScript)
+	if err != nil {
+		framework.Failf("unable to locate %q: %v", webhookDeployScript, err)
+	}
+
+	fmw := framework.NewDefaultFramework("fpgaplugin-e2e")
+
+	ginkgo.It("Run FPGA plugin tests", func() {
+		// Run region test case twice to ensure that device is reprogrammed at least once
+		runTestCase(fmw, pluginYamlPath, webhookDeployScriptPath, serviceAccountYamlPath, "region", "orchestrated", arria10dcp12Resource, nlb3PodResource, 3, 0)
+		runTestCase(fmw, pluginYamlPath, webhookDeployScriptPath, serviceAccountYamlPath, "region", "orchestrated", arria10dcp12Resource, nlb0PodResource, 0, 3)
+		// Run af test case
+		runTestCase(fmw, pluginYamlPath, webhookDeployScriptPath, serviceAccountYamlPath, "af", "preprogrammed", nlb0NodeResource, nlb0PodResource, 0, 3)
+	})
+}
+
+func runTestCase(fmw *framework.Framework, pluginYamlPath, webhookDeployScriptPath, serviceAccountYamlPath, pluginMode, webhookMode, nodeResource, podResource string, nlb1, nlb2 int16) {
+	ginkgo.By(fmt.Sprintf("deploying webhook in %s mode", webhookMode))
+	_, _, err := framework.RunCmd(webhookDeployScriptPath, "--mode", webhookMode, "--namespace", fmw.Namespace.Name)
+	framework.ExpectNoError(err)
+
+	ginkgo.By("creating service account for the plugin")
+	framework.RunKubectlOrDie("apply", "-f", serviceAccountYamlPath)
+
+	ginkgo.By(fmt.Sprintf("setting %s mode for the FPGA plugin", pluginMode))
+	annotation := fmt.Sprintf("fpga.intel.com/device-plugin-mode=%s", pluginMode)
+	framework.RunKubectlOrDie("annotate", "node", "--overwrite", "--all", annotation)
+
+	ginkgo.By("deploying FPGA plugin")
+	framework.RunKubectlOrDie("apply", "-f", pluginYamlPath)
+
+	ginkgo.By("waiting for FPGA plugin's availability")
+	if _, err := e2epod.WaitForPodsWithLabelRunningReady(fmw.ClientSet, "kube-system",
+		labels.Set{"app": "intel-fpga-plugin"}.AsSelector(), 1, 10*time.Second); err != nil {
+		framework.DumpAllNamespaceInfo(fmw.ClientSet, fmw.Namespace.Name)
+		framework.LogFailedContainers(fmw.ClientSet, fmw.Namespace.Name, framework.Logf)
+		framework.Failf("unable to wait for all pods to be running and ready: %v", err)
+	}
+
+	resource := v1.ResourceName(nodeResource)
+	ginkgo.By("checking if the resource is allocatable")
+	if err := utils.WaitForNodesWithResource(fmw.ClientSet, resource, 30*time.Second); err != nil {
+		framework.Failf("unable to wait for nodes to have positive allocatable resource: %v", err)
+	}
+
+	resource = v1.ResourceName(podResource)
+	image := "intel/opae-nlb-demo:devel"
```

Comment:
I believe we'd also want to be able to test release images. Perhaps we should add simple kustomization for `opae-nlb-demo` image too? (for reference, I've hacked something for `crypto-perf` modes).

Reply:
That would be nice to have, agree. Should it be added to [this issue]https://github.com/intel/intel-device-plugins-for-kubernetes/issues/318)?

Reply:
I've added a bullet for it.

Reply:
ok, thanks